packages/expo-cli/src/credentials/views/__tests__/AndroidKeystore-Remove-test.ts

```diff
@@ -0,0 +1,105 @@
+import fs from 'fs-extra';
+import { getCtxMock, testExperienceName } from '../../test-fixtures/mocks-android';
+import { RemoveKeystore } from '../AndroidKeystore';
+import prompt from '../../../prompt';
+import { mockExpoXDL } from '../../../__tests__/mock-utils';
+
+jest.mock('../../actions/list');
+jest.mock('../../../prompt');
+jest.mock('fs-extra');
+mockExpoXDL({
+  AndroidCredentials: {
+    generateUploadKeystore: jest.fn().mockImplementation(() => ({
+      keystorePth: 'test',
+      keystorePassword: 'test',
+      keyAlias: 'test',
+      keyPassword: 'test',
+    })),
+  },
+});
+
+const originalWarn = console.warn;
+const originalLog = console.log;
+beforeAll(() => {
+  console.warn = jest.fn();
+  console.log = jest.fn();
+});
+afterAll(() => {
+  console.warn = originalWarn;
+  console.log = originalLog;
+});
+beforeEach(() => {
+  (prompt as any).mockReset();
+  (prompt as any).mockImplementation(() => {
+    throw new Error('Should not be called');
+  });
+});
+
+describe('RemoveKeystore', () => {
+  describe('existing credentials', () => {
+    it('should display warning prompt and abort', async () => {
+      const ctx = getCtxMock();
+      (prompt as any)
+        .mockImplementationOnce(() => ({ confirm: false })) // prompt with warning message, abort
+        .mockImplementationOnce(x => {
+          throw new Error("shouldn't happen");
+        });
+
+      const view = new RemoveKeystore(testExperienceName);
+      const lastView = await view.open(ctx);
+
+      expect(lastView).toBe(null);
+      expect((prompt as any).mock.calls.length).toBe(1);
+      expect(ctx.android.fetchKeystore.mock.calls.length).toBe(1);
+      expect(ctx.android.removeKeystore.mock.calls.length).toBe(0);
+    });
+
+    it('should display warning prompt and continue', async () => {
+      const ctx = getCtxMock();
+      fs.pathExists.mockImplementationOnce(() => false);
+
+      // first: prompt with warning message, true means continue
+      // second: ask if cli should display credentials, user won't see that because when() should return false
+      (prompt as any)
+        .mockImplementationOnce(() => ({ confirm: true }))
+        .mockImplementationOnce(question => {
+          if (question.when()) {
+            throw new Error("shouldn't happen");
+          }
+          return { confirm: undefined };
+        })
+        .mockImplementationOnce(() => {
```

Comment:
```suggestion
        .mockImplementation(() => {
```

Reply:
Please make the same change for every other mocked implementation with `throw new Error("shouldn't happen");`. If the number of prompts changes we will notice this error (`shouldn't happen`) instead of some timeout.

Reply:
On the other hand, I think it doesn't matter - the first call will throw the error so the code won't get to the later stage.